programs/mango-v4/src/state/orderbook/book.rs

```diff
@@ -114,6 +114,10 @@ impl<'a> Orderbook<'a> {
         let opposing_bookside = self.bookside_mut(other_side);
         for best_opposing in opposing_bookside.iter_all_including_invalid(now_ts, oracle_price_lots)
         {
+            if remaining_base_lots == 0 || remaining_quote_lots == 0 {
```

Comment:
either I can do this, or the even less invasive change would be to only skip the insertion of 0 match_base_lots even to event queue part

Reply:
This exist condition is good. With this, we can also remove the `if done { break }` and the whole `done` variable later in the function.

Reply:
done